src/mbox/message-box.lisp

```diff
@@ -179,6 +198,23 @@ This function sets the result as `handler-result' in `item'. The return of this
               (bt2:condition-notify withreply-cvar)))
           (handler-fun)))))
 
+
+(declaim (ftype (function (message-box/bt)
+                          (values &optional))
+                ensure-thread-is-running))
+
+(defun ensure-thread-is-running (msgbox)
+  (with-slots (queue-thread thread-lock)
+      msgbox
+    (bt2:with-lock-held (thread-lock)
+      (unless (bt2:thread-alive-p queue-thread)
+        (log:trace "Restarting thread ~A"
```

Comment:
I'd like to have this as a `warn` logging to know exactly when it happens. Should be an exception.